### DISCUSSION TITLE
Is there anyway to apply MaxQueryDepthInstrumentation in each query in a given GraphQL request

### DISCUSSION THREAD

**Discussion by adimulam** [graphql-java/graphql-java#2795] [General]
I have a GraphQL Request with multiple queries in it (Not Batching, just different queries sent in same single request).

Currently MaxQueryDepthInstrumentation is applied on the entire GraphQL Request. But each query may have different max-depth-allowed. Is there anyway to apply MaxQueryDepthInstrumentation on each query in the request? Allow the query to resolve if it is within max-depth and if there are any queries that exceed it's max-depth, that particular query should be aborted. Hence allowing partial results (i.e, some queries resolve and some fail, but all within the same single request).

**Comment by bbakerman** [comment]
> Is there anyway to apply MaxQueryDepthInstrumentation in each query in a given GraphQL request

No - what you want seems pretty custom - I would copy the `MaxQueryDepthInstrumentation` and create your own `CustomMaxQueryDepthInstrumentation`

**Reply by adimulam** [reply]
Thanks for the reply @bbakerman . 

Catch here is, we should still allow partial results i.e, the queries within the depth-limit should still get resolved and the queries that exceed should get aborted. All within the same single GraphQL request.

with CustomMaxQueryDepthInstrumentation, I can somehow validate each query's depth in a given GraphQL request.
But how do we modify the original query-string (i.e, remove the query that exceeded depth-limit). Or is there any way to set the exception on those particular in-validated queries and let graphql-java continue to resolve the valid depth queries.

Thanks.
